### ISSUE TITLE
Add field for Request Stop to public transportation station presets

### ISSUE THREAD

**Issue by 1ec5** [openstreetmap/id-tagging-schema#875]
### OSM Tag(s)

request_stop=yes

### How would you like this tag to see supported?

As a Field

### Label

Request Stop

### Aliases

Flag Stop, Whistle Stop

### Terms

stop on request

### Link to OSM Wiki page

https://wiki.openstreetmap.org/wiki/Tag:request_stop%3Dyes

### Status of the Tag

In Use

### Usage of the tag

3,210

### Further Information

Historically, some regions used `railway=halt` to distinguish request stops from other railway stations. However, this usage conflicts with a different definition used in German-speaking countries to distinguish sites with and without rail switches, and yet another definition in North America to distinguish any station that has a relatively limited service schedule. Additionally, some bus systems also use a request-stop model extensively.

`request_stop=yes` allows the request-stop aspect of the stop to be tagged explicitly and unambiguously, regardless of the mode of transportation. It gained currency sometime in 2022, particularly in Europe, where `railway=halt` is also common, though there’s a lot of potential for this tag to be used in North America as well.

**Comment by matkoniecz** [comment]
Note that `request_stop=yes` is tricky in cases where some routes stop always and some on demand.

**Comment by matkoniecz** [comment]
It seems to remain unclear - I will mark it as blocked by lack of a clear documentation

**Comment by 1ec5** [comment]
> Note that `request_stop=yes` is tricky in cases where some routes stop always and some on demand.

That would be the subject of [this proposal](https://community.openstreetmap.org/t/rfc-feature-proposal-role-request-stop/135656) that stalled due to many other nuances around stop requests. Use of `request_stop=yes` continues unabated however.

> It seems to remain unclear - I will mark it as blocked by lack of a clear documentation

Can you clarify what else needs to be [documented](https://wiki.openstreetmap.org/wiki/Tag:request_stop%3Dyes) for this field to be considered?

**Comment by matkoniecz** [comment]
clear description how cases of  stop where some routes stop always and some on demand should be handled

(and ideally also when basically all are on demand but few stop always and opposite)

**Comment by 1ec5** [comment]
That sounds like a tagging question rather than a documentation issue. As far as I know, there isn’t any agreement about how to handle that edge case. But there are clearly plenty of cases where it’s clearer.

**Comment by matkoniecz** [comment]
My strategy would be to reject tags (especially lower-importance ones) with unresolved tagging and documentation issues.

Especially more fundamental ones. I think that it should be at the very least documented that lack of clarity. Not sure whether it should be treated as a fundamental blocker.

**Comment by 1ec5** [comment]
The same issue equally applies to the previous tag for this situation, `railway=halt`, which does have a preset. 🤔

**Comment by matkoniecz** [comment]
I guess that missing used preset is a bigger problem than missing used attribute?

(also it was not me who added `railway=halt` - I cannot promise being 100% consistent with decisions taken year before)

**Comment by 1ec5** [comment]
Oh yeah, I’m not blaming you. My understanding is that `request_stop=yes` is intended to break out one of the historic meanings of `railway=halt` more explicitly. It started out as a compromise in the longstanding dispute between English-speaking and German-speaking countries over the meaning of `railway=halt`.

A couple years ago, Polish mappers raised a concern on the forum that a given bus stop could be a request stop for some routes but not others. I’m even familiar with some examples of this locally. However, I suspect we can assume that it’s far less common among railway stations and halts, due to the operating realities of a train line.

**Comment by matkoniecz** [comment]
> A couple years ago, Polish mappers raised a concern on the forum that a given bus stop could be a request stop for some routes but not others

maybe it was even me :)

**Comment by Geo-2695** [comment]
In my region, `railway=halt` is used for small stations with no ticket office, no switches, and only passenger boarding/alighting services. Some scheduled trains stop here regularly, while others pass through without stopping. None of these stops operate as request stops.
 
Once a station has switches, regardless of its operational status, it is tagged as `railway=station`.

**Comment by matkoniecz** [comment]
I will close it as blocked by unresolved documentation issues. Noting at minimum that it is unsolved problem is needed, though maybe also solving it is needed (tag neither?).

**Comment by 1ec5** [comment]
I’d be happy to resolve the documentation issue, but I’m at a loss as to what needs to be documented. `request_stop=*` is originally about rail stations. Is mixed request/non-request state even a problem for those, or is this a theoretical concern from potentially extending the key to bus stops? The key seems to satisfy all the [general guidelines](https://github.com/openstreetmap/id-tagging-schema/blob/main/GUIDELINES.md#general-guidelines). If the concern over mixed state is so great that it shouldn’t even be presented for rail stations, maybe the field’s location set could exclude Poland?

**Comment by 1ec5** [comment]
I [rewrote the documentation](https://wiki.openstreetmap.org/wiki/Tag:request_stop%3Dyes) based on my understanding of the situation. Please correct anything that seems wrong or missing.

Looking into the statistics, I think the good news is that the concern about mixed-rules stops is completely unfounded in countries that primarily use `request_stop=yes` on railway features. The bad news is that there’s still some regional fragmentation: unlike other countries, the U.S. is only using the tag on `railway=stop`, a usage that’s completely undocumented. This surprised me, since I remember when `request_stop=yes` was the reason we gave up on `railway=halt` in favor of `railway=station`. I think this happened because we used to tag `railway=halt` at the stopping position, not where `railway=station` is mapped today. `request_stop=yes` must have slipped through the cracks as the stations were upgraded to PTv2.

Do you think this discrepancy is enough to block a `request_stop=*` checkbox in other countries that use the tag as it has been documented, such as the UK and Germany?

**Comment by CAM-Gerlach** [comment]
I mostly map bus rather than rail lines (as I live in a part of the US where passenger rail is entirely nonexistent), some of which have "request stops". While it is touched on by the current guidance, it could perhaps be a little more explicit about what specific non-trail PT objects using the modern public transport schema should be tagged `request_stop=*`. I.e. does this only go on `public_transport=platform`? `public_transport=stop`? Both? Is this the same for bus, tram or other modes or does it differ (given they historically have differed in what is actually tagged as the `highway=bus_stop`? FWIW, The mention of non-rail use is so relatively brief, whereas the coverage of railway use is by comparison so comprehensive (going into considerable historical and practical detail), that I actually missed it the former entirely on my first read through the article and only spotted it at all when looking through a second time writing this comment--so adding a bit more non-rail content would be helpful, I think :).

Sidenote--while I would personally aver this should not be closed at all, the close reason was given as "completed" rather than "not planned", which seems likely to have been a mistake (although close reasons were first implemented [over four years ago](https://github.blog/changelog/2022-05-19-the-new-github-issues-may-19th-update/#%F0%9F%95%B5%F0%9F%8F%BD%E2%99%80%EF%B8%8F-issue-closed-reasons), I still see mistakes with them from time to time).

**Comment by 1ec5** [comment]
The page is intended to be descriptive, not prescriptive. That would be the job of a tagging proposal or at least a forum thread. Unfortunately, I didn’t have much to say about the tag’s use on bus infrastructure because it’s so rare or unspoken of in the part of the world where I map. But it is very common in some European countries, so I hope participants in this thread who hail from there can pitch in. 🙂 In the meantime, I think it’s always been possible to split this discussion into adding a checkbox for rail presets versus bus presets, which are harder.

**Comment by matkoniecz** [comment]
> I’m at a loss as to what needs to be documented

At least that it is unclear what should be done where some lines just stop and some need request,  ideally  with link to discussion(s) where it was discussed

**Comment by matkoniecz** [comment]
Re: network analysis.  Note it is a bit optimistic that in given network all routes do the same and that all relevant routes are mapped

**Comment by matkoniecz** [comment]
But either way main blocker as I have seen it is removed. 

As this is mostly exotic oddity it should not block support,  I think.

**Comment by matkoniecz** [comment]
> the close reason was given as "completed" rather than "not planned", which seems likely to have been a mistake (although close reasons were first implemented [over four years ago](https://github.blog/changelog/2022-05-19-the-new-github-issues-may-19th-update/#%F0%9F%95%B5%F0%9F%8F%BD%E2%99%80%EF%B8%8F-issue-closed-reasons),

I am using close/ open as binary state and not use this github feature, in the same way as type field is not used and labels are used instead (partially because I treat it as vendor lock-in, partially because interface is bad,  partially because it is ovecomplicated)

**Comment by 1ec5** [comment]
> Re: network analysis. Note it is a bit optimistic that in given network all routes do the same and that all relevant routes are mapped

Yes, it’s only a first-order approximation to give a sense of perspective. I think I demonstrated that this is very unlikely to be a major issue within the parameters I specified, because a transit agency needs to be able to effectively communicate the rules to their customers. Moreover, I have an unproven hunch that we’re using `request_stop=*` to indicate whether a stop request is always required at the stop. In other words, if the stop gets mixed-request service, then we don’t tag it as a request stop. I would need to gain more confidence in that hypothesis before documenting it though.

**Comment by matkoniecz** [comment]
I would not say so. Poland has official bus stops operated by city used also by small private bus companies that at lower end may be single person operating a single vehicle. 

And these small buses often are stopping solely and only on demand. While more official ones stop always. And their status is tagged, ignoring these extra users.  Who may or may have not used these bus stops illegally.

**Comment by 1ec5** [comment]
I was speaking about the U.S. and UK, where I did that analysis for the wiki page. I’m aware that Poland uses this tag extensively on shared bus stops and it’s a sore point there.
